tools/codegen/core/gen_experiments.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
```

Comment:
how will this file be invoked? would it need to be done by the user? We should include some docstring here with info on adding an experiment.

Reply:
done